### ISSUE TITLE
Reduce the amount of monkeypatching we do

### ISSUE THREAD

**Issue by AlexWaygood** [PyCQA/flake8-pyi#183]
I have been [explicitly warned](https://github.com/PyCQA/flake8/issues/1560#issuecomment-1036341793) that the next release of flake8 is likely to break flake8-pyi, due to the amount of monkeypatching we do in this plugin, and a refactoring of flake8 internals that's currently underway. We should probably do something about that. 

Ideally, I think we'd file PRs to flake8 and pyflakes so that they become pyi-compatible out of the box, rendering some of the crazy hacks in our code base unnecessary. I'm not really sure where to start, though.

**Comment by JelleZijlstra** [comment]
Yes, ideally flake8 should have native support for linting pyi files, and then this plugin can focus just on providing pyi-specific error codes, instead of the crazy monkeypatching we do right now.

**Comment by AlexWaygood** [comment]
fixed by https://github.com/PyCQA/flake8-pyi/pull/530
